aws-java-sdk-core/src/test/java/com/amazonaws/http/conn/ssl/TestSSLSocket.java

```diff
@@ -15,13 +15,45 @@
 package com.amazonaws.http.conn.ssl;
 
 import java.io.IOException;
+import java.util.Arrays;
 
 import javax.net.ssl.HandshakeCompletedListener;
 import javax.net.ssl.SSLSession;
 import javax.net.ssl.SSLSocket;
 
 class TestSSLSocket extends SSLSocket {
 
+    @Override
+    public void setEnabledProtocols(String[] protocols) {
```

Comment:
I agree with the rationale behind this change, I think it improves the test quality. I'm thinking that it would be better to have TestSSLSocket just capture the protocols passed to setEnabledProtocols and then the test code can query that and make assertions based on that. That gives you a bit more flexibility on what you can assert on since it moves the assertion out of this mock object and into the respective tests.

So something like

TestSSLSocket socket = ...
f.prepareSocket(socket)
Assert.arrayEquals(socket.getCapturedProtocols(), expectedProtocols);